packages/graphql-live-query-patch/src/createLiveQueryPatchDeflator.ts

```diff
@@ -0,0 +1,63 @@
+import type { LiveExecutionResult } from "@n1ru4l/graphql-live-query";
+import { compare } from "fast-json-patch";
+import { ExecutionResult } from "graphql";
+import { LiveExecutionPatch } from "./LiveExecutionPatch";
+
+export async function* createLiveQueryPatchDeflator(
+  asyncIterator: AsyncIterableIterator<LiveExecutionResult>
+): AsyncIterableIterator<LiveExecutionPatch & ExecutionResult> {
+  let previousValue: LiveExecutionResult["data"] | null = null;
+  let revision = 0;
+  for await (const value of asyncIterator) {
+    // no live query? reset state.
+    if (!value.isLive) {
```

Comment:
add missing data check here

Reply:
@D1plo1d  So the idea is that a live query implementation adds the `isLive` property to the execution result, so we can identify a live query and then apply the json patch transform.

Any other ideas that don't require adding the property?

Ideally, I would like to apply the patch deflator to any execution result and just pass through non-live query results and then specifically handle the live query results.

Reply:
@n1ru4l It's a tricky situation for sure. The only other option I can think of would be to add a graphql-internal double underscored `__isLive` property to resolver results.

Cons:

- you have to iterate through all the top level fields to determine if data is live
- not sure how to implement an `__isLive` on the resolver results. I'm guessing the directive can modify resolver results?

Pros:

- doesn't require adding a property to the async iterator
- there aren't usually many top level fields in a query* so I'd expect the performance difference is negligible (*but that's only in my experience)
- shouldn't conflict with user defined field names because double underscore is reserved for GraphQL internal use

```
expect(value).toEqual({
    value: {
      data: {
        foo: {
          __isLive: true,
          bar: "kek",
        },
      },
    },
    done: false,
  });
```

Reply:
I think you might confusing adding isLive to the execution result vs adding it on the asynciterator. 

Currently a live directive can also only be applied on the query definition, not for individual fields

Reply:
> I think you might confusing adding isLive to the execution result vs adding it on the asynciterator.
Yup XP

> Currently a live directive can also only be applied on the query definition, not for individual fields
Ok, fair so it would be:
```js
expect(value).toEqual({
    value: {
      data: {
        __isLive: true,
        foo: {
          bar: "kek",
        },
      },
    },
    done: false,
  });
```

The only advantage I'd see on doing that vs the execution result would be that in future it could help with live directives on fields but I'm thinking that's a premature optimization on my part and easy to change later.

Current implementation gets my :+1: